packages/admin/admin/src/dateTime/TimePicker.tsx

```diff
@@ -0,0 +1,202 @@
+import { createComponentSlot } from "../helpers/createComponentSlot";
+import { type ThemedComponentBaseProps } from "../helpers/ThemedComponentBaseProps";
+
+export type Future_TimePickerClassKey = "root" | "clearInputAdornment" | "readOnlyAdornment" | "openPickerAdornment" | "openPickerButton";
+
+export type Future_TimePickerProps = ThemedComponentBaseProps<{
+    root: typeof MuiTimePicker<Date, true>;
+    clearInputAdornment: typeof CometClearInputAdornment;
+    readOnlyAdornment: typeof InputAdornment;
+    openPickerAdornment: typeof InputAdornment;
+    openPickerButton: typeof IconButton;
+}> & {
+    fullWidth?: boolean;
+    required?: boolean;
+    value?: string;
+    onChange?: (time: string | undefined) => void;
+    iconMapping?: {
+        openPicker?: React.ReactNode;
+        readOnly?: React.ReactNode;
+    };
+} & Omit<MuiTimePickerProps<Date, true>, "value" | "onChange" | "slotProps">;
+
+const getTimeString = (date: Date) => {
+    return format(date, "HH:mm");
+};
+
+const getDateFromTimeString = (value: string | undefined): Date | null => {
+    if (!value) {
+        return null;
+    }
+
+    const parsedDate = parse(value, "HH:mm", new Date());
+    const isValid = !isNaN(parsedDate.getTime());
+
+    if (!isValid) {
+        throw new Error(`Invalid time value: "${value}", must be a 24h time in format HH:mm`);
+    }
+
+    return parsedDate;
+};
+
+export const Future_TimePicker = (inProps: Future_TimePickerProps) => {
+    const {
+        iconMapping = {},
+        fullWidth,
+        required,
+        slotProps,
+        disabled,
+        value: stringValue,
+        onChange,
+        readOnly,
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminFutureTimePicker",
+    });
+    const [open, setOpen] = useState(false);
+    const dateValue = getDateFromTimeString(stringValue);
+
+    const { openPicker: openPickerIcon = <Time color="inherit" />, readOnly: readOnlyIcon = <Lock fontSize="inherit" /> } = iconMapping;
+
+    const clearButtonEndAdornment =
+        !required && !disabled && !readOnly ? (
+            <ClearInputAdornment
+                position="end"
+                hasClearableContent={Boolean(dateValue)}
+                onClick={() => onChange?.(undefined)}
+                {...slotProps?.clearInputAdornment}
+            />
+        ) : null;
+
+    const readOnlyAdornment = readOnly ? (
+        <ReadOnlyAdornment position="end" {...slotProps?.readOnlyAdornment}>
+            {readOnlyIcon}
+        </ReadOnlyAdornment>
+    ) : null;
+
+    const openPickerAdornment = (
+        <OpenPickerAdornment position="start" {...slotProps?.openPickerAdornment}>
+            <OpenPickerButton onClick={() => setOpen(true)} disabled={disabled || readOnly} {...slotProps?.openPickerButton}>
+                {openPickerIcon}
+            </OpenPickerButton>
+        </OpenPickerAdornment>
+    );
```

Comment:
Shouldn't these be wrapped in useMemo to avoid unnecessary rerenders?

Reply:
I created separate components for `OpenPickerAdornment` and `ReadOnlyAdornment`, also I refactored the way `ClearInputAdornment ` is rendered, so this should be solved. I also made those changes in the existing `Future_DatePicker`. 

(See the updated PR description)